src/shared-api-objects/current-user.ts

```diff
@@ -0,0 +1,23 @@
+import { BehaviorSubject } from "rxjs";
+import { openmrsObservableFetch, FetchResponse } from "../openmrs-fetch";
+import { mergeAll, filter, map } from "rxjs/operators";
+
+const userSubject = new BehaviorSubject(
+  openmrsObservableFetch("/ws/rest/v1/session")
+);
+
+export function getCurrentUser(opts: CurrentUserOptions = {}) {
+  return userSubject.asObservable().pipe(
+    mergeAll(),
+    map((r: FetchResponse) => (opts.includeAuthStatus ? r.data : r.data.user)),
+    filter(Boolean)
```

Comment:
ensuring we only produce a value to the observer when the value is truthy. The value will be falsy when `r.data.user` is undefined (because the user is not logged in)